src/doc/src/reference/config.md

```diff
@@ -238,13 +238,27 @@ precedence rules as other options specified directly with `--config`.
 
 ### Config-relative paths
 
-Paths in config files may be absolute, relative, or a bare name without any
-path separators. Paths for executables without a path separator will use the
-`PATH` environment variable to search for the executable. Paths for
-non-executables will be relative to where the config value is defined. For
-config files, that is relative to the parent directory of the `.cargo`
-directory where the value was defined. For environment variables it is
-relative to the current working directory.
+Paths in config files may be absolute, relative, or a bare name without any path separators.
+Paths for executables without a path separator will use the `PATH` environment variable to search for the executable. 
+Paths for non-executables will be relative to where the config value is defined.
+In particular, rules are:
+
+* For environment variables it is relative to the current working directory.
+* For config files, that is relative to the parent directory of the directory where the value was defined.
```

Comment:
Should we also explicitly call out `--config key=value`?

Reply:
I'm not seeing the change for this

Reply:
I thought you were suggesting something like this, no?

https://github.com/rust-lang/cargo/pull/11079/files#diff-0917d2867ed0f7116151e8c28138a91760a17f3fca809cc72523053c2762f3afR248-R250

Reply:
I feel like I'm missing something.

I see `config.md` covering
- Environment variables
- Implicit config files
- Explicit config files via `--config <path>`

I'm not seeing `--config key=value` being discussed for where paths would be relative to.

Reply:
Current implementation as a reference.
https://github.com/rust-lang/cargo/blob/c1871146e3f5f5f5913a3ef93ca176b062d7ff3e/src/cargo/util/config/value.rs#L66-L75